lib/math/hash.flow

```diff
@@ -1,21 +1,30 @@
 // © Copyright 2011 Area9 Technologies.
-import runtime;
-import string;
+import math/bits;
+import text/binary;
 
 export {
 	hashString(key : string) -> int;
 	// Faster version of hash for strings, uses tail recursion
 	fastHashString(key : string) -> int;
 	// Even faster version of hash for strings: uses only 'cutoff' symbols from the string
 	fastHashStringWithCutoff(key : string, cutoff : int) -> int;
+
+	// Constants for FNV hash: http://en.wikipedia.org/wiki/Fowler_Noll_Vo_hash
+	FNV_offset_basis = 0x811C9DC5;
+	FNV_prime = 16777619;
+
+	// Update hash with integer/double/string value
+	addHashInt(hash: int, v: int) -> int;
+	addHashDouble(hash: int, d: double) -> int;
```

Comment:
Add a comment that this allocates and is likely slow.

Reply:
Done.